tests/test_jn.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2020 D-Wave Systems Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import nbformat
+from nbconvert.preprocessors import ExecutePreprocessor
+import unittest
+
+def run_jn(jn, timeout):
+    
+    open_jn = open(jn, "r", encoding='utf-8')
+    notebook = nbformat.read(open_jn, nbformat.current_nbformat)
+    open_jn.close()
+        
+    preprocessor = ExecutePreprocessor(timeout=timeout, kernel_name='python3')
+    preprocessor.allow_errors = True    
+    preprocessor.preprocess(notebook, {'metadata': {'path': os.path.dirname(jn)}})
+
+    return notebook
+
+def collect_jn_errors(nb):
+
+    errors = []
+    for cell in nb.cells:
+        if 'outputs' in cell:
+            for output in cell['outputs']:
+                if output.output_type == 'error':
+                    if output.evalue == 'no embedding found':
+                        return ["Embedding failed"]
+                    else:
+                        errors.append(output)
+
+    return errors
+
+def robust_run_jn(jn, timeout, retries):
+
+    run_num = 1
+    notebook = run_jn(jn, timeout)
+    errors = collect_jn_errors(notebook)
+
+    while errors == ['Embedding failed'] and run_num < retries:
```

Comment:
... and then test errors like this:
```suggestion
    while 'no embedding found' in errors and run_num < retries:
```

Reply:
Nice, done, thanks!